garak/payloads.py

```diff
@@ -172,49 +169,63 @@ def _scan_payload_dir(self, dir) -> dict:
     def _refresh_payloads(self) -> None:
         """Scan resources/payloads and the XDG_DATA_DIR/payloads for
         payload objects, and refresh self.payload_list"""
-        self.payload_list = self._scan_payload_dir(PAYLOAD_DIR)
+        self.__class__.payload_list = self._scan_payload_dir(PAYLOAD_DIR)
```

Comment:
With the current definition of `payload_list` being class level and not in `__init__()` this is equivalent:
```suggestion
        self.payload_list = self._scan_payload_dir(PAYLOAD_DIR)
```

Reply:
I don't think class attributes assigned through instantiation are mutable from instances

Reply:
You are correct although this is somewhat of a grey area, attributes cannot be replaced via instance access but are mutable themselves. In this case `None` is being replaced with a pointer to the resulting array. However the resulting object can be modified.

This particular line does need to access `self.__class__.payload_list` however the `read` access in `search()` does not need to use `__class__`, I am on the fence on if those locations *should* use `__class__` for clarity reasons.

Reply:
I'm on the fence too. I guess py's "explicit not implicit" suggests leaving it in, and one can find a rationale for that - the verbose formulation signals that there might be a non-trivial access pattern. Also doing nothing is usually the lighter path.